### DISCUSSION TITLE
Access to an undefined property App\Models\User::$lower_en_name.

### DISCUSSION THREAD

**Discussion by waelsjaber** [larastan/larastan#1253] [Q&A] [answered]
When I run `larastan` on code contains  `select from DB` query, I got an issue as below:

```
Access to an undefined property App\Models\User::$lower_en_name.
```

My code is:
```php

class UserController extends Controller
{

    public function index(Request $request)
    {
    
        $users = User::select(['id', 'type', DB::raw("lower(name->>'en') as lower_en_name")])->get();

        return response()->json($users);
    }
    
 }
```

How is the best way to solve this issue?

**Comment by canvural** [comment, marked as answer]
Hi,

This is not an easy task to understand for a static analysis.

Best way to solve is ignore these kind (property coming from raw sql select) of errors.

**Reply by szepeviktor** [reply]
For some people static analysis is so valuable that they write code that can be statically analyzed.

**Reply by waelsjaber** [reply]
ok, thank you
